File: python/silapay_connector/silapay_connector/providers/mercadopago.py
import os


class MercadoPagoProvider:
    def __init__(self, access_token=None):
        self.access_token = access_token or os.getenv("MERCADOPAGO_ACCESS_TOKEN")

    def pix(self, dados):
        raise NotImplementedError(
            "MercadoPago: metodo pix() ainda nao implementado. Consulte https://www.mercadopago.com.br/developers"
        )

    def boleto(self, dados):
        raise NotImplementedError(
            "MercadoPago: metodo boleto() ainda nao implementado. Consulte https://www.mercadopago.com.br/developers"
        )

    def cartao(self, dados):
        raise NotImplementedError(
            "MercadoPago: metodo cartao() ainda nao implementado. Consulte https://www.mercadopago.com.br/developers"
        )

    def saldo(self):
        raise NotImplementedError(
            "MercadoPago: metodo saldo() ainda nao implementado. Consulte https://www.mercadopago.com.br/developers"
        )

File: typescript/silapay-connector/src/providers/mercadopago.ts
import type { ConnectorOptions, PaymentProvider } from "../types";

export class MercadoPagoProvider implements PaymentProvider {
    private readonly accessToken?: string;

    constructor(options: ConnectorOptions = {}) {
        this.accessToken = options.accessToken || process.env.MERCADOPAGO_ACCESS_TOKEN;
    }

    async pix(): Promise<never> {
        throw new Error(
            "MercadoPago: método pix() ainda não implementado. Consulte https://www.mercadopago.com.br/developers"
        );
    }

    async boleto(): Promise<never> {
        throw new Error(
            "MercadoPago: método boleto() ainda não implementado. Consulte https://www.mercadopago.com.br/developers"
        );
    }

    async cartao(): Promise<never> {
        throw new Error(
            "MercadoPago: método cartao() ainda não implementado. Consulte https://www.mercadopago.com.br/developers"
        );
    }

    async saldo(): Promise<never> {
        throw new Error(
            "MercadoPago: método saldo() ainda não implementado. Consulte https://www.mercadopago.com.br/developers"
        );
    }
}
